types/src/crypto/asymmetric_key/tests.rs

```diff
@@ -0,0 +1,42 @@
+use crate::{crypto::SecretKey, PublicKey};
+
+#[test]
+fn can_construct_ed25519_keypair_from_zeroes() {
+    let bytes = [0; SecretKey::ED25519_LENGTH];
+    let secret_key = SecretKey::ed25519(bytes);
+    let _public_key: PublicKey = secret_key.into();
+}
+
+// is this correct?
```

Comment:
Yes, it seems so.  Setting the secret key to all zeros hits [this code](https://github.com/RustCrypto/traits/blob/elliptic-curve-v0.5.0/elliptic-curve/src/scalar.rs#L32-L38), and from there we get to [this code](https://github.com/CasperLabs/casper-node/blob/a31d1edd4a1067993f51eb40027c3a504dee00b8/types/src/crypto/asymmetric_key.rs#L365).

The first linked code block is the only way `k256::PublicKey::from_secret_key` can fail I believe, although possibly the `.expect` warrants a comment to explain why it's safe.

I believe it's not a problem in production as there should be no reasonable way for a non-malicious user to have a secret key which is all zeros.  They can't construct one, and they won't ever receive one from a malicious peer.

Reply:
Agreed.  Had also verified this after I wrote that, just forgot to remove the comment.